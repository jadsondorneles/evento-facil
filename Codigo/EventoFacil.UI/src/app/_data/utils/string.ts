export const ehString = (input: unknown): input is string => {
  return typeof input === 'string'
}
