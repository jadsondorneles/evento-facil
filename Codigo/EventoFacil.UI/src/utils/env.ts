import { PUBLIC_ENV_KEY } from 'next-runtime-env'

import { isClient } from '@/app/_data/utils/browser'

/**
 * Lê uma variável de ambiente segura do navegador ou de qualquer ambiente
 * variável do servidor (process.env) definida no arquivo @see {@link ./app/layout.tsx}.
 *
 * Uso:
 * ```ts
 * const API_URL = env('API_URL')
 * ```
 */
export function env(key: string): string | undefined {
  if (isClient()) {
    return window[PUBLIC_ENV_KEY]?.[key] || ''
  }
  return process.env[key]
}
