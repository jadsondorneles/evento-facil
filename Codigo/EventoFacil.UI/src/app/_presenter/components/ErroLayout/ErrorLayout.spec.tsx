import mockRouter from 'next-router-mock'

import '@testing-library/jest-dom'
import { fireEvent, screen } from '@testing-library/react'

import { renderizarComTema } from '@/jest.utils'

import animationData from './Erro404.json'
import ErrorLayout from './index'

// Mock do componente Lottie
jest.mock('react-lottie', () => {
  return function DummyLottie() {
    return <div data-testid='lottie-mock'></div>
  }
})

jest.mock('next/navigation', () => require('next-router-mock'))

describe('ErrorLayout', () => {
  const renderComponent = () => {
    return renderizarComTema(
      <ErrorLayout
        animationData={animationData}
        titulo='Ocorreu um erro!!'
        detalheErro='Erro: Descrição da mensagem de erro'
      />
    )
  }

  it('deve corresponder ao snapshot', () => {
    const component = renderComponent()
    expect(component).toMatchSnapshot()
  })

  describe('ao clicar no botão Voltar', () => {
    it('redireciona o usuário para a página anterior', () => {
      renderComponent()
      const button = screen.getByRole('button')
      expect(button).toBeInTheDocument()
      expect(fireEvent.click(button!)).toBeTruthy()

      expect(mockRouter).toMatchObject({
        pathname: '/',
      })
    })
  })
})
