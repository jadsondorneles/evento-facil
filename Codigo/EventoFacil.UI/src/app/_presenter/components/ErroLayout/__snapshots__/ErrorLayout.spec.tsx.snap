// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorLayout deve corresponder ao snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-t1otnx"
      >
        <div
          class="MuiBox-root css-18nw4yk"
        >
          <div
            class="MuiBox-root css-u4kl8w"
          >
            <div
              data-testid="lottie-mock"
            />
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ykj3ce-MuiTypography-root"
          >
            Ocorreu um erro!!
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1un1obg-MuiTypography-root"
          >
            Erro: Descrição da mensagem de erro
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-ahei1p-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ChevronLeftOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
            Voltar para a Página Anterior
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-t1otnx"
    >
      <div
        class="MuiBox-root css-18nw4yk"
      >
        <div
          class="MuiBox-root css-u4kl8w"
        >
          <div
            data-testid="lottie-mock"
          />
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ykj3ce-MuiTypography-root"
        >
          Ocorreu um erro!!
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1un1obg-MuiTypography-root"
        >
          Erro: Descrição da mensagem de erro
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-ahei1p-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronLeftOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
          Voltar para a Página Anterior
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
